fastq/fastqBig.go

```diff
@@ -36,7 +38,45 @@ func ToFastqBig(a *Fastq) *FastqBig {
 	copy(answer.SeqRc, answer.Seq)
 	dna.ReverseComplement(answer.SeqRc)
 	answer.Qual = a.Qual
-	answer.Rainbow = dnaTwoBit.NewTwoBitRainbow(answer.Seq)
-	answer.RainbowRc = dnaTwoBit.NewTwoBitRainbow(answer.SeqRc)
+	answer.Rainbow = dnaTwoBit.TwoBitRainbowDeReference(answer.Seq)
+	answer.RainbowRc = dnaTwoBit.TwoBitRainbowDeReference(answer.SeqRc)
 	return answer
 }
+
+// ReadFqBig returns a FastqBig struct that is converted to dnaTwoBit format
+// and includes a rainbow offset to perform hash look ups in GSW aligner.
+func ReadFqBig(reader *fileio.SimpleReader) (*FastqBig, bool) {
+	answer := FastqBig{}
+	line, done := fileio.ReadLine(reader)
+	if done {
+		return nil, true
+	}
+	answer.Name = strings.Split(string(line.String()[1:]), " ")[0]
+	line, done = fileio.ReadLine(reader)
+	if done {
+		return nil, true
+	}
+	//set up sequence and reverse comp
+	answer.Seq = dna.ByteSliceToDnaBases(line.Bytes())
+	answer.SeqRc = make([]dna.Base, len(answer.Seq))
+	copy(answer.SeqRc, answer.Seq)
+	dna.ReverseComplement(answer.SeqRc)
+
+	//performs two bit conversion
+	answer.Rainbow = dnaTwoBit.TwoBitRainbowDeReference(answer.Seq)
+	answer.RainbowRc = dnaTwoBit.TwoBitRainbowDeReference(answer.SeqRc)
+
+	line, done = fileio.ReadLine(reader)
+	if done {
+		return nil, true
```

Comment:
Should checks like this be an error instead of returning nil and true?  Seems like this would mean we were reading a fq file that had not been fully written.